serde-generate/runtime/golang/src/lcs/deserializer.go

```diff
@@ -0,0 +1,198 @@
+// Copyright (c) Facebook, Inc. and its affiliates
+// SPDX-License-Identifier: MIT OR Apache-2.0
+
+package lcs
+
+import (
+	"bytes"
+	"errors"
+
+	"github.com/facebookincubator/serde-reflection/serde"
```

Comment:
It's golang's convention to use a uri to import your package. See https://golang.org/doc/code.html#ImportingRemote .

It has huge benefit, to give you example, I updated my https://github.com/xli/serde-reflection master branch with mod name "github.com/xli/serde-reflection/serde-generate/runtime/golang/src" (see https://github.com/xli/serde-reflection/commit/de760e8ba0674649b2558ce7e91d1b434e87b043)

Then in my libra go codebase import "github.com/xli/serde-reflection/serde-generate/runtime/golang/src/lcs",
and use the lcs.
Without doing anything else, when running test / build, I got:

```bash
go test -run='' .
go: finding module for package github.com/xli/serde-reflection/serde-generate/runtime/golang/src/lcs
go: downloading github.com/xli/serde-reflection v0.0.0-20200815182109-5a0de5262f9e
go: downloading github.com/xli/serde-reflection/serde-generate/runtime/golang/src v0.0.0-20200815182109-5a0de5262f9e
go: found github.com/xli/serde-reflection/serde-generate/runtime/golang/src/lcs in github.com/xli/serde-reflection/serde-generate/runtime/golang/src v0.0.0-20200815182109-5a0de5262f9e
go: downloading github.com/xli/serde-reflection v0.0.0-20200815025256-253d68995d66
ok  	github.com/libra/libra-client-sdk-go/libraclient	(cached)
```

It means, we don't need install this runtime when generating code, we just need import correct path "github.com/facebookincubator/serde-reflection/serde-generate/runtime/golang/src/serde" in generated code, it will just work.
The test golang_generation change proves this.

I have not remove the "src" yet, but if we all agree this is better way to install the runtime, we should remove the "src" under golang.

----------------

Install as serde is not worth it, because it is customized GOPATH, user will run into various problems, and they can't checkin the code into their go repository, because in general, people follow the golang guide to create repository under xxx.com/repo/....
Hence if copy to the repo, it means the serde should be imported as "xxx.com/repo/serde", and it won't work for our own code import "serde".
This will become nightmare for user, as even the data structures are exactly same, but golang consider they are different types, because the import path is different.

Reply:
Thanks for the explanations. For users (I mean external repos) indeed it is great if we can use a native package manager and avoid "source install". Brandon, Yucong, and I talked about packaging the python language runtime at some point and deprecate "source install". (It looks like Cmake or Maven also have a notion of external project)

What I don't understand is: when we develop in serde-reflection, how do we make sure that "lcs" refer to the module "serde" of the same branch (PR or local) of the same git repo (github official, forked, or local)?

Reply:
it will refer to same version because they are both under github.com/facebookincubator/serde-reflection/serde-generate/runtime/golang/src, notice the actual download is "github.com/facebookincubator/serde-reflection", and the other parts are under the repo, hence you always get same version, although my go.mod is inside "github.com/facebookincubator/serde-reflection/serde-generate/runtime/golang/src", that's just for local development to recognize the path when my actual repo is at ~/code/serde-reflection/, when go run test / build, it find go.mod, and consider the module name is used instead of path.

Reply:
Ok so it looks the information we are looking for is in https://golang.org/doc/gopath_code.html (and chapter 10 of the Go book). My understanding is that one needs both (1) suitable urls in package names (for users) and (2) setting GOPATH for local development.

Reply:
Yes for (1), not sure what you mean (2), GOPATH should be set by user as they need. 
For our development, we follow go modules https://blog.golang.org/using-go-modules, go modules are default after go 1.13 release, and it is addition to GOPATH for local development, not necessary required for production as all dependencies can still be pull in by go get into GOPATH, and behave same.

Looks like we have agreement to not use "serde" as import path, I removed "src" under golang dir.

Reply:
I think we are converging but let me schedule a meeting tomorrow to iron out the details